report_templates/host_compare_content_hosts_packages.erb

```diff
@@ -16,40 +16,38 @@ template_inputs:
   resource_type: Katello::ActivationKey
 model: ReportTemplate
 -%>
-<%- packages = [] -%>
-<%- host1 = load_hosts(search: "name = #{input('Host 1')}").first.first -%>
-<%- host2 = load_hosts(search: "name = #{input('Host 2')}").first.first -%>
-<%- report_headers 'Package', host1.name, host2.name, 'Architecture', 'Status' -%>
-<%- set1 = host1.installed_packages.group_by { |p| p.name } -%>
-<%- set2 = host2.installed_packages.group_by { |p| p.name } -%>
-<%- (set1.keys | set2.keys).uniq.each do |package_name| -%>
-<%-   packages << { 'Package' => package_name,
-                     host1.name => (set1.key?(package_name) ? set1[package_name].first.nvra : '-'),
-                     host2.name => (set2.key?(package_name) ? set2[package_name].first.nvra : '-') } -%>
-<%- end -%>
-<%- packages.each do |data| -%>
-<%-   version1 = data[host1.name].gsub(data["Package"] + "-", "").gsub(/\.el[0-9][^.]*/, "").gsub(/\.[n|x|i].*/, "") -%>
-<%-   version2 = data[host2.name].gsub(data["Package"] + "-", "").gsub(/\.el[0-9][^.]*/, "").gsub(/\.[n|x|i].*/, "") -%>
-<%-   arch_list = [ data[host1.name].split('.').last, data[host2.name].split('.').last ].uniq -%>
-<%-   arch_list.delete("-") -%>
-<%-   if version1 == "-" -%>
-<%-     status = "#{host2.name} only"-%>
-<%-   elsif version2 == "-" -%>
-<%-     status = "#{host1.name} only"-%>
-<%-   elsif version(version1) > version(version2) -%>
-<%-     status = "greater in #{host1.name}"-%>
-<%-   elsif version(version1) < version(version2) -%>
-<%-     status = "lower in #{host1.name}"-%>
-<%-   else  -%>
-<%-     status = "same version"-%>
-<%-   end -%>
-<%-   report_row(
-        'Package': data['Package'],
-        "#{host1.name}": data[host1.name],
-        "#{host2.name}": data[host2.name],
-        'Architecture': arch_list[0],
-        'Status': status
-      )
--%>
-<%- end -%>
-<%= report_render -%>
+<%
```

Comment:
why do you switch to single erb tag? if you want to start a new best practice, please update all other reports, however I don't think it's actually better

also I believe this does not work, report_render needs to be in `<%=` so that it actually prints the report

Reply:
The missing `<%=` was a typo thanks for that.

I won't be fixing all templates in this patch, sorry. A single tag is million times more friendly to edits than what is the "best practice", but it is solely my opinion and I accept that other people might like different things.